src/main/java/frc/robot/subsystems/climber/Climber.java

```diff
@@ -3,79 +3,132 @@
+
     private final UnitModel unitModel = new UnitModel(Constants.Climber.TICKS_PER_RAD);
 
+    private final ArmFeedforward feedforward = new ArmFeedforward(Constants.Climber.F_FORWARD_S, Constants.Climber.F_FORWARD_COS, Constants.Climber.F_FORWARD_V, Constants.Climber.F_FORWARD_A);
+
+    private final PIDController m_controller = new PIDController(Constants.Climber.P_VELOCITY, Constants.Climber.I_VELOCITY, Constants.Climber.D_VELOCITY);
+
+    private final Encoder m_encoder = new Encoder(Ports.Climber.ENCODER_A_CHANNEL, Ports.Climber.ENCODER_B_CHANNEL);
+    private final DCMotor m_armGearbox = DCMotor.getFalcon500(2);
+
+    private final SingleJointedArmSim m_armSim =
+            new SingleJointedArmSim(
+                    m_armGearbox,
+                    Constants.Climber.GEAR_RATIO,
+                    SingleJointedArmSim.estimateMOI(Constants.Climber.ARM_LENGTH, Constants.Climber.ARM_MASS),
+                    Constants.Climber.ARM_LENGTH,
+                    Units.degreesToRadians(Constants.Climber.MIN_ANGLE),
+                    Units.degreesToRadians(Constants.Climber.MAX_ANGLE),
+                    Constants.Climber.ARM_MASS,
+                    true,
+                    VecBuilder.fill(Constants.Climber.ARM_ENCODER_DIST_PER_PULSE)
+            );
+
+    private final EncoderSim m_encoderSim = new EncoderSim(m_encoder);
+
+    private final Mechanism2d m_mech2d = new Mechanism2d(60, 60);
+    private final MechanismRoot2d m_armPivot = m_mech2d.getRoot("ArmPivot", 30, 30);
+    private final MechanismLigament2d m_armTower =
+            m_armPivot.append(new MechanismLigament2d("ArmTower", 30, -90));
+
+    private final MechanismLigament2d m_arm =
+            m_armPivot.append(
+                    new MechanismLigament2d(
+                            "Arm",
+                            30,
+                            Units.radiansToDegrees(m_armSim.getAngleRads()),
+                            6,
+                            new Color8Bit(Color.kYellow)));
+
+
     public Climber() {
+        if (Robot.isSimulation()) {
+            SmartDashboard.putData("Arm sim", m_mech2d);
+            m_encoder.setDistancePerPulse(Constants.Climber.ARM_ENCODER_DIST_PER_PULSE);
+            m_armTower.setColor(new Color8Bit(Color.kBlue));
+        }
 
-        /*
-         Set the left motor on Brake mode.
-         */
-        leftMotor.setNeutralMode(NeutralMode.Brake);
 
         /*
-         sets the phase of the sensor
+         set the right motor on Brake mode.
          */
-        leftMotor.setSensorPhase(Ports.Climber.LEFT_SENSOR_PHASE);
+        mainMotor.setNeutralMode(NeutralMode.Brake);
 
         /*
-         checking is motor inverted.
+        sets the phase of the sensor.
          */
-        leftMotor.setInverted(Ports.Climber.IS_LEFT_INVERTED);
+        mainMotor.setSensorPhase(Ports.Climber.MAIN_SENSOR_PHASE);
```

Comment:
I only found out today, but you don't need to set the sensor phase of Falcons ([reference](https://phoenix-documentation.readthedocs.io/en/latest/ch14_MCSensor.html#sensor-phase)).
Also, I recommend using `TalonFXInvertType`  for the `setInverted` as it's more descriptive (I only found out about it today)